core/src/tile/tileManager.cpp

```diff
@@ -407,9 +410,13 @@ void TileManager::updateTileSet(TileSet& _tileSet, const ViewState& _view) {
             if (entry.getProxyCounter() > 0) {
                 if (entry.isReady()) {
                     m_tiles.push_back(entry.tile);
-                } else if (curTileId.z < maxZoom) {
-                    // Cancel loading
-                    removeTiles.push_back(curTileId);
+                } else if (entry.isInProgress()) {
+                    if (curTileId.z >= maxZoom || curTileId.z <= minZoom) {
+                        // Cancel tile loading but keep tile entry for referencing
+                        // this tiles proxy tiles.
+                        _tileSet.source->cancelLoadingTile(*entry.task);
```

Comment:
Benefit of having a proxy tile is only when it is actually available to be rendered, since it is not in this `else if` branch, I think we can just cancel loading and remove this in progress proxyTile, in favor of spending resources for the actual visible tileset?

Reply:
Ohh I think I remember this now, even though this tile is loading, this tile's proxy tiles will also be removed if we remove this tile, hence we can not remove this tile!